Source/WebGPU/WebGPU/RenderBundleEncoder.mm

```diff
@@ -210,16 +231,27 @@
     // MTLIndirectCommandBuffers don't support debug commands.
 }
 
+static void addResource(RenderBundle::ResourcesContainer* resources, id<MTLResource> mtlResource, ResourceUsageAndRenderStage *resource)
+{
+    if (ResourceUsageAndRenderStage *existingResource = [resources objectForKey:mtlResource]) {
+        existingResource.usage |= resource.usage;
+        existingResource.renderStages |= resource.renderStages;
+    } else
+        [resources setObject:resource forKey:mtlResource];
+}
+
+static void addResource(RenderBundle::ResourcesContainer* resources, id<MTLResource> mtlResource, MTLRenderStages stage)
+{
+    return addResource(resources, mtlResource, [[ResourceUsageAndRenderStage alloc] initWithUsage:MTLResourceUsageRead renderStages:stage]);
+}
+
 void RenderBundleEncoder::setBindGroup(uint32_t groupIndex, const BindGroup& group, uint32_t dynamicOffsetCount, const uint32_t* dynamicOffsets)
 {
     UNUSED_PARAM(dynamicOffsetCount);
     UNUSED_PARAM(dynamicOffsets);
 
     id<MTLIndirectRenderCommand> icbCommand = currentRenderCommand();
     if (!icbCommand) {
-        if (group.vertexArgumentBuffer())
```

Comment:
We always have to set `m_icbDescriptor.maxVertexBufferBindCount` to the count from the device (which is usually 24) otherwise the generated shader buffer offsets will be incorrect